app/src/main/java/org/kiwix/kiwixmobile/zim_manager/fileselect_view/effects/DeleteFiles.kt

```diff
@@ -40,18 +40,31 @@ data class DeleteFiles(private val booksOnDiskListItem: List<BookOnDisk>) :
 
   override fun invokeWith(activity: AppCompatActivity) {
     (activity as ZimManageActivity).cachedComponent.inject(this)
+
+    var name = "\n"
     booksOnDiskListItem.forEach {
-      dialogShower.show(DeleteZim(it), {
+      name += "\n" + it.book.title
+    }
+
+    dialogShower.show(DeleteZims(name), {
+      var allDeleted = true
```

Comment:
It would also be for the best if this var were a val, here I think a fold will work best, if you start with an initial value of true and in the accumulator && with the result of `deleteSpecificZimFile(it)` then by the end if there has been a failure the value will be false

Reply:
@macgills I have a bit of a doubt here. `deleteSpecificZimFile(it)` will be called each time the forEach loop runs. I can't declare a val inside the loop. If I do that, then I won't be able to use the val outside the loop for performing the check.

Reply:
I was suggesting getting rid of the loop and using a fold expression.

I'll give a small example of folding by writing a quick scratch


![image](https://user-images.githubusercontent.com/3358282/78904699-d9608d80-7a74-11ea-8d9d-e7b887e81042.png)


You give fold the initial thing you want to start with and an operation to perform that must return the same type as the initial thing. The function gets called for every item in the list and each successive call to `{acc, element -> if(i<3) acc+element else acc }` uses the result of the previous invocation eg
```
1st item
acc= emptyList , element  =1, 1<3 so add it to acc
2nd item
acc= listOf(1), element  =2, 2<3 so add it to acc
3rd item
acc= listOf(1,2), element  =3, 3 is not <3 so just return acc
```

Reply:
@macgills thanks for the information :)
I misinterpreted the earlier comment and confused myself.